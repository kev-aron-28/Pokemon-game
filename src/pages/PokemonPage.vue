<template>
  <h1 v-if="!pokemon">Cargando...</h1> 
  <div v-else>
    <h1>Quien es el pokemon?</h1>
  
    <pokemon-picture 
      :pokemonId="pokemon.id" 
      :showPokemon="showPokemon"/>
      
      <pokemon-options 
        :pokemons="pokemonArr"
        @selection="checkAnswer($event)"/>
      
      <div v-if="showAnser">
         <h2>{{message}}</h2>
        <button @click="newGame">Nuevo juevo</button>

      </div>
       
  </div>
  
   
</template>

<script>
import PokemonPicture from '../components/PokemonPicture.vue';
import PokemonOptions from '../components/PokemonOptions.vue'
import getPokemonOptions from '../helpers/getPokemonOptions.js'

export default {
  name: 'PokemonPage',
  components: {
    PokemonPicture,
    PokemonOptions
  },

  data(){
    return {
      pokemonArr: [],
      pokemon: null,
      showPokemon: false,
      showAnser: false,
      message: ''
    }
  },

  methods: {
   async mixPokemonArr(){
     this.pokemonArr = await getPokemonOptions()
     const rndInt = Math.floor( Math.random() * 4 )
     this.pokemon = this.pokemonArr[rndInt]
   },

    checkAnswer( pId ){
      this.showPokemon = true
      this.showAnser = true
      if( pId === this.pokemon.id ){
        this.message = `Correcto`
      } else {
        this.message = `Oops, era ${this.pokemon.name}`
      }
    },

    newGame(){
      this.showPokemon = false;
      this.showAnser = false;
      this.pokemonArr = []
      this.pokemon = null
      this.mixPokemonArr()
    }
  },
  
  mounted() {
    this.mixPokemonArr()
  }

}
</script> 

<style>

</style>
