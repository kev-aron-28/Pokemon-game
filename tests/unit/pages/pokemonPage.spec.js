import { shallowMount, mount } from '@vue/test-utils'
import PokemonPage from '@/pages/PokemonPage'
import {pokemons} from '../mocks/pokemons.mock'
describe('PokemonPage component ', () => {
		
	let wrapper
	beforeEach(() => {
		wrapper = shallowMount( PokemonPage )
 
	})

	test('debe de hacer match con el snapshot', () => {
		expect(wrapper.html()).toMatchSnapshot()
	})
	
	test('debe de llamar el mixed pokemon al montar', () => {
		
		const mixPokemonArrSpy = jest.spyOn(PokemonPage.methods, 'mixPokemonArr')	
		const wrapper = shallowMount(PokemonPage)
		expect(mixPokemonArrSpy).toHaveBeenCalled()

	})


	test('debe de hacer match con el snapshot cuando carga los pokemons', () => {
		const wrapper = mount(PokemonPage, {
			data(){
				return {
					pokemonArr: pokemons,
      		pokemon: pokemons[0],
      		showPokemon: false,
      		showAnser: false,
      		message: ''
				}
			}
		})
		
		expect( wrapper.html() ).toMatchSnapshot()
	})

	test('debe de mostrar los componentes de pokemonPicture y pokemonOptions', () => {
			const wrapper = shallowMount(PokemonPage, {
				data(){
					return {
					pokemonArr: pokemons,
      		pokemon: pokemons[0],
      		showPokemon: false,
      		showAnser: false,
      		message: ''
				}
			}
		})

		const picture = wrapper.find('pokemon-picture-stub')
		const options = wrapper.find('pokemon-options-stub')
		expect( picture.exists() ).toBeTruthy()
		expect( options.exists() ).toBe(true)

		expect( picture.attributes('pokemonid') ).toBe('1')
		expect( options.attributes('pokemons') ).toBeTruthy()
	})

	test('pruebas en el checkanswer respuesta correcta', async () => {
		const wrapper = shallowMount(PokemonPage, {
				data(){
					return {
					pokemonArr: pokemons,
      		pokemon: pokemons[0],
      		showPokemon: false,
      		showAnser: false,
      		message: ''
				}
			}
		})
		
		await wrapper.vm.checkAnswer(1)

		expect( wrapper.find('h2').exists() ).toBeTruthy()
		expect( wrapper.find('h2').text() ).toBe('Correcto')
		expect( wrapper.vm.showPokemon ).toBeTruthy()
	})


	test('Pruebas en el checkanswer respuesta incorrecta', async () => {
		const wrapper = shallowMount(PokemonPage, {
    	data() {
      	return {
        	pokemonArr: pokemons,
        	pokemon: pokemons[0],
        	showPokemon: false,
        	showAnser: false,
        	message: "",
      	}
    	},
  	});

		await wrapper.vm.checkAnswer(2)
		
		expect( wrapper.vm.message ).not.toBe('Correcto')

	})

})
