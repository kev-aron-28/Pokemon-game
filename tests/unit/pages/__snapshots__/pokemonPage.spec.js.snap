// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage component  debe de hacer match con el snapshot 1`] = `<h1>Cargando...</h1>`;

exports[`PokemonPage component  debe de hacer match con el snapshot cuando carga los pokemons 1`] = `
<div>
  <h1>Quien es el pokemon?</h1>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon" class="hidden-pokemon">
    <!--v-if-->
  </div>
  <div class="options-container">
    <ul>
      <li>bulbasaur</li>
      <li>ivysaur</li>
      <li>venusaur</li>
      <li>charmander</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
